src/Node/ForNode.php

```diff
@@ -101,7 +101,11 @@ public function compile(Compiler $compiler): void
         $compiler->write("\$_parent = \$context['_parent'];\n");
 
         // remove some "private" loop variables (needed for nested loops)
-        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\'], $context[\'loop\']);'."\n");
+        $compiler->write('unset($context[\'_seq\'], $context[\'_iterated\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\']);'."\n");
+
+        if ($this->getAttribute('with_loop')) {
+            $compiler->write('unset($context[\'loop\']);'."\n");
```

Comment:
Can we use a single `unset()` call in this case?

Reply:
Sure, done!